ocean_lib/web3_internal/contract_handler.py

```diff
@@ -136,14 +191,32 @@ def _load(contract_name, address=None):
             address = contract_definition.get("address")
             assert address, "Cannot find contract address in the abi file."
             address = Web3.toChecksumAddress(address)
+        assert address is not None, "address shouldn't be None at this point"
 
         abi = contract_definition["abi"]
         bytecode = contract_definition["bytecode"]
         contract = Web3Provider.get_web3().eth.contract(
             address=address, abi=abi, bytecode=bytecode
         )
+        if contract.address is None:  # if web3 drops address, fix it
+            contract.address = address
+        assert contract.address is not None
+
         ContractHandler._set(contract_name, contract)
-        return ContractHandler._contracts[(contract_name, address)]
+
+        # postconditions
+        name = contract_name
+        (contract1, concise_contract1) = ContractHandler._contracts[name]
+        assert contract1 is not None
```

Comment:
these assertions seem duplicated from _set(). Is there any way we can extract them into a function?

Reply:
Good point. Fixed in 83c7a5b